packages/data-provider/src/utils/types.ts

```diff
@@ -327,24 +335,34 @@ type Mutable<T> = {
 };
 
 export type MutableDataDiff = {
-	old: Mutable<DataRecord>;
-	new: Mutable<DataRecord>;
+	old: MutableDataRecord;
+	new: MutableDataRecord;
 };
 
+export type MutableDataRecord = Mutable<DataRecord>;
```

Comment:
type is used before being declared. TS doesn't care but it reads poorly. Don't worry about changing this, there is no need for Mutable type after the lectern client upgrade.

Reply:
This should be handled in next PR